core/src/services/s3/backend.rs

```diff
@@ -116,6 +117,13 @@ impl S3Builder {
         self
     }
 
+    /// Set bucket versioning status of this backend
+    pub fn bucket_versioning_enabled(mut self, enabled: bool) -> Self {
```

Comment:
Following the same API naming pattern, we could call it `enable_versioning`. Also, it's better to move this API to later part.